node/tests/ScanTest.test.ts

```diff
@@ -376,6 +378,166 @@ describe("Scan GlideClusterClient", () => {
+                        while (!cursor.isFinished()) {
+                            result = await testClient.scan(cursor);
+                            cursor = result[0];
+                        }
+
+                        // If scan completes without error, wait and retry
+                        await new Promise((resolve) =>
+                            setTimeout(resolve, 1000),
+                        );
+                    } catch (error) {
+                        if (
+                            error instanceof Error &&
+                            error.message.includes(
+                                "Could not find an address covering a slot, SCAN operation cannot continue",
+                            )
+                        ) {
+                            // Expected error occurred
+                            errorReceived = true;
+                        } else {
+                            // Unexpected error, rethrow
+                            throw error;
+                        }
+                    }
+                }
+
+                expect(errorReceived).toBe(true);
+
+                // Perform scan with 'allowNonCoveredSlots: true'
+                cursor = new ClusterScanCursor();
+
+                while (!cursor.isFinished()) {
+                    result = await testClient.scan(cursor, {
+                        allowNonCoveredSlots: true,
+                    });
+                    cursor = result[0];
+                }
+
+                expect(cursor.isFinished()).toBe(true);
+
+                // Get keys using 'KEYS *' from the remaining nodes
+                const keys: GlideString[] = [];
+
+                for (const address of allOtherAddresses) {
+                    const result = await testClient.customCommand(
+                        ["KEYS", "*"],
+                        {
+                            route: {
+                                type: "routeByAddress",
+                                host: address[0],
+                                port: address[1],
+                            },
+                        },
+                    );
+                    keys.push(...(result as GlideString[]));
+                }
+
+                // Scan again with 'allowNonCoveredSlots: true' and collect results
+                cursor = new ClusterScanCursor();
+                const results: GlideString[] = [];
+
+                while (!cursor.isFinished()) {
+                    result = await testClient.scan(cursor, {
+                        allowNonCoveredSlots: true,
+                    });
+                    results.push(...result[1]);
+                    cursor = result[0];
+                }
+
+                // Compare the sets of keys obtained from 'KEYS *' and 'SCAN'
+                expect(new Set(results)).toEqual(new Set(keys));
+            } finally {
+                testClient.close();
+                await testCluster.close();
```

Comment:
Just to confirm - does it stop the cluster?

Reply:
yep